lib/std/build.zig

```diff
@@ -168,6 +169,11 @@ pub const Builder = struct {
     }
 
     pub fn destroy(self: *Builder) void {
+        switch (self.vcpkg_root) {
+            .Found => |path| self.allocator.free(path),
+            else => {},
+        }
+
```

Comment:
This is actually not needed; all this code already depends on an arena allocator.

Reply:
Since the whole build package does not bother with freeing, no problem, but I'm not convinced by this way of doing.
What's the point of allowing any allocator type as input if the code assumes it's an arena and does not respect the usual allocation contract that you need to free memory ? We might as well modify the file to only deal with `ArenaAllocator`.

Reply:
Modifying the file to only deal with ArenaAllocator would be reasonable, although I suspect it would result in a lot of `&arena.allocator` usage due to calling non-build-system std lib APIs that are more general purpose and correctly deal with allocators, including freeing.

I think that the code is doing what you are saying already, by having a reference to `Builder` and reaching into it to use its arena allocator everywhere. It's just more convenient to access the `b.allocator` rather than `&b.arena.allocator`.